pandas/tseries/index.py

```diff
@@ -740,6 +740,18 @@ def _add_offset(self, offset):
                            PerformanceWarning)
             return self.astype('O') + offset
 
+    def putmask(self, mask, value):
+        """
```

Comment:
Yes, shoud I include the logic to ``putmask``? I think it is better, but I concerned it may break something.